Add plugin: YTSummarizer

ardakalayci opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin: https://github.com/ardakalayci/ytsummarizer

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [x]  macOS
  - [x]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

joethei commented:
A few screenshots / GIF's showcasing the plugin functionality in the README would be great.

What sets your plugin apart from the "Youtube Summarizer" / "YouTube Video Summarizer"  plugins?

[Copyright (c) 2023 Łukasz Strzępek](https://github.com/ardakalayci/ytsummarizer/blob/f951a57c4149a980fe8a7953c9bbf25f769b2550/LICENSE#L3)
Update this.

Also it looks like you copied a lot of code from the "YTranscript" plugin, but you don't credit the original author anywhere in your README, please do so.

["description": "A plugin that fetches YouTube transcripts and generates summaries using OpenAI GPT's",](https://github.com/ardakalayci/ytsummarizer/blob/f951a57c4149a980fe8a7953c9bbf25f769b2550/manifest.json#L6)
Avoid including sentences like "This is a plugin that does ..." in your description, it should be a given since this is only shown when browsing plugins through the app or on our website.
We have published a guide for plugin descriptions to follow here: https://docs.obsidian.md/Plugins/Releasing/Submission+requirements+for+plugins#Keep+plugin+descriptions+short+and+simple.

["commands": [](https://github.com/ardakalayci/ytsummarizer/blob/f951a57c4149a980fe8a7953c9bbf25f769b2550/manifest.json#L10-L21)
This is not a valid part of the manifest, remove it.

[await this.app.vault.modify(file, content);](https://github.com/ardakalayci/ytsummarizer/blob/f951a57c4149a980fe8a7953c9bbf25f769b2550/src/main.ts#L149), [await this.app.vault.modify(file, contentWithLoadingMessage);](https://github.com/ardakalayci/ytsummarizer/blob/f951a57c4149a980fe8a7953c9bbf25f769b2550/src/main.ts#L159), [await this.app.vault.modify(file, updatedContent);](https://github.com/ardakalayci/ytsummarizer/blob/f951a57c4149a980fe8a7953c9bbf25f769b2550/src/main.ts#L188)
[Prefer `Vault.process` instead of `Vault.modify` to modify a file in the background](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Prefer+%60Vault.process%60+instead+of+%60Vault.modify%60+to+modify+a+file+in+the+background)
Please note that this requires that you increase your `minAppVersion` to at least 1.7.2

[containerEl.createEl("h2", { text: "Settings for YTranscript" });](https://github.com/ardakalayci/ytsummarizer/blob/f951a57c4149a980fe8a7953c9bbf25f769b2550/src/main.ts#L230)
[Don't add a top-level heading in the settings tab, such as "General", "Settings", or the name of your plugin.](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Only+use+headings+under+settings+if+you+have+more+than+one+section.)

[containerEl.createEl("h3", { text: "OpenAI Settings" });](https://github.com/ardakalayci/ytsummarizer/blob/f951a57c4149a980fe8a7953c9bbf25f769b2550/src/main.ts#L273)
For section headings in settings use:
```ts
new Setting(containerEl).setName('name here').setHeading();
```

[containerEl.createEl("h3", { text: "OpenAI Settings" });](https://github.com/ardakalayci/ytsummarizer/blob/f951a57c4149a980fe8a7953c9bbf25f769b2550/src/main.ts#L273)
[Avoid "settings" in settings headings](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Avoid+%22settings%22+in+settings+headings)

[.setName("OpenAI Model")](https://github.com/ardakalayci/ytsummarizer/blob/f951a57c4149a980fe8a7953c9bbf25f769b2550/src/main.ts#L289), [.setName("Custom Summary Prompt")](https://github.com/ardakalayci/ytsummarizer/blob/f951a57c4149a980fe8a7953c9bbf25f769b2550/src/main.ts#L303), [.setName("Max Tokens")](https://github.com/ardakalayci/ytsummarizer/blob/f951a57c4149a980fe8a7953c9bbf25f769b2550/src/main.ts#L317), [sidebarButton.setButtonText("Show in Sidebar")](https://github.com/ardakalayci/ytsummarizer/blob/f951a57c4149a980fe8a7953c9bbf25f769b2550/src/prompt-modal.ts#L55), [newPageButton.setButtonText("Create New Page")](https://github.com/ardakalayci/ytsummarizer/blob/f951a57c4149a980fe8a7953c9bbf25f769b2550/src/prompt-modal.ts#L62), [summaryButton.setButtonText("Generate Summary");](https://github.com/ardakalayci/ytsummarizer/blob/f951a57c4149a980fe8a7953c9bbf25f769b2550/src/transcript-view.ts#L226), [this.videoTitle || "YouTube Video",](https://github.com/ardakalayci/ytsummarizer/blob/f951a57c4149a980fe8a7953c9bbf25f769b2550/src/transcript-view.ts#L256), [this.summaryContainerEl.createEl("p", { text: "Generating summary..." });](https://github.com/ardakalayci/ytsummarizer/blob/f951a57c4149a980fe8a7953c9bbf25f769b2550/src/transcript-view.ts#L245), [text: "Video Summary",](https://github.com/ardakalayci/ytsummarizer/blob/f951a57c4149a980fe8a7953c9bbf25f769b2550/src/transcript-view.ts#L263), [copyButton.setButtonText("Copy to Clipboard");](https://github.com/ardakalayci/ytsummarizer/blob/f951a57c4149a980fe8a7953c9bbf25f769b2550/src/transcript-view.ts#L286)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

[sidebarButton.buttonEl.addClass("mod-cta");](https://github.com/ardakalayci/ytsummarizer/blob/f951a57c4149a980fe8a7953c9bbf25f769b2550/src/prompt-modal.ts#L54)
Use `sidebarButton.setCta()` instead.

ardakalayci commented:
@joethei  Thank you for your detailed review and feedback on the YTSummarizer plugin. I've addressed all the issues you mentioned:

1. **Screenshots/GIFs**: Added multiple screenshots to the README showcasing the plugin's functionality.

2. **Plugin differentiation**: Added a new section "What sets YTSummarizer apart?" to clearly explain how this plugin differs from other YouTube-related plugins, focusing on AI-powered summaries, flexible viewing options, interactive timestamps, and customizable experience.

3. **Copyright information**: Updated the LICENSE file with the correct copyright information (Arda Kalaycı, 2024).

4. **Attribution to YTranscript**: Added proper attribution to the YTranscript plugin by Łukasz Strzępek in the README's Acknowledgments section, clearly stating that YTSummarizer builds upon YTranscript's functionality.

5. **Plugin description**: Shortened the description and removed "This is a plugin that..." as recommended.

6. **Invalid manifest section**: Removed the invalid "commands" section from manifest.json.

7. **Vault.process usage**: Replaced all instances of `Vault.modify` with `Vault.process` and updated minAppVersion to 1.7.2 accordingly.

8. **Settings headings**: Removed the top-level heading in settings and replaced h3 headings with `setHeading()` as recommended.

9. **UI text case**: Updated all UI text to use sentence case instead of title case.

10. **Button styling**: Replaced `buttonEl.addClass("mod-cta")` with `setCta()`.

I believe all the issues have been addressed according to Obsidian's plugin guidelines. The plugin is now ready for another review. Thank you for your guidance in helping improve the quality of this plugin.

joethei commented:
[await this.app.vault.process(file, () => content);](https://github.com/ardakalayci/ytsummarizer/blob/b57ff74fe5bc5ae29bbe84980671a9ec9bd9c512/src/main.ts#L149)
This is not how you use the `process` function, take a look at this YouTube video: https://youtu.be/RvtV9DKr3ag?t=52
